### PR TITLE
fix(invity): backup warning

### LINKED ISSUE #2967: Exchange unexpected warning
1. take device without backup
1. go to exchanges tab
1. click 1/4 amount

observe 
image

note it does not trigger any on device dialog.

### PR DESCRIPTION
**Pull request by vladimirvolek**
This resolves: https://github.com/trezor/trezor-suite/issues/2967

### CODE REVIEW

**Comment by mroz22** [comment]
Maybe it would be better to extend backup validation logic inside Trezor Connect to be able to suppress emit of this event. It would be done somewhere here I guess https://github.com/trezor/connect/blob/83620a24e2600c55d4a5f385701cceff9098147c/src/js/core/Core.js#L459

**Comment by martinboehm** [comment]
@szymonlesisz could you please evaluate the change in Connect as proposed by @mroz22?

**Comment by mroz22** [comment]
Also changes I proposed should not be allowed in popup mode I'd say.

**Comment by szymonlesisz** [comment]
i agree that it should be changed in connect, but i dont have a time for this right now.
however proposed change will not work. modal will not be displayed but the connect is still awaiting for user confirmation.
there is more elegant way how to ignore this request:
https://github.com/trezor/trezor-suite/blob/develop/packages/suite/src/middlewares/wallet/discoveryMiddleware.ts#L26-L35

by swallowing this action in the middleware and not populate it to reducer (by `next(action);`)

**Comment by mroz22** [comment]
>  connect is still awaiting for user confirmation

Even when param `showOnTrezor: false`? 

And sorry beforehead for being lazy to read it in the code :sloth:

**Comment by szymonlesisz** [comment]
yes, this is potential coin loss operation and its not parameterized (maybe it should for trusted host)
tldr; it's basically a artificial button request

**Comment by martinboehm** [comment]
@szymonlesisz OK, how about merging the PR as it is for now. The behaviour without the fix is confusing for the user.
The change is very small and when you develop a more proper version in Connect, it can be easily changed.

File: packages/suite/src/components/suite/modals/index.tsx [hunk 1]
```diff
@@ -116,12 +114,16 @@ const getDeviceContextModal = (props: Props) => {
...
-
+    const { modal, modalActions, goto, router } = props;
     if (modal.context !== MODAL.CONTEXT_DEVICE_CONFIRMATION) return null;
+    const isCoinmarketExchange = router.route?.name === 'wallet-coinmarket-exchange';
 
     switch (modal.windowType) {
         case 'no-backup':
+            // do not show this modal in exchanges - calling getAddress() just for address placeholder
+            // useCoinmarketExchangeForm (line 104)
+            if (isCoinmarketExchange) return null;
```
Comment (tomexx) [inline packages/suite/src/components/suite/modals/index.tsx]: I'd consider creating a new switch case option and use that instead of creating a special case here. But that's just a thing to consider as I don't have clear picture of the architecture yet.

Comment (martinboehm) [reply packages/suite/src/components/suite/modals/index.tsx]: @tomexx would you be please so kind and show us what you have in mind in code

Comment (tomexx) [reply packages/suite/src/components/suite/modals/index.tsx]: When I'm looking at it again I see I formulated it incorrectly. My point was if it isn't better to do the decision whether to show or not to show the confirm even before calling this function (= where you call the function) instead of returning `null` here. Or create another switch param for this case and use it where needed. 

But as I wrote before I did not study the architecture of modals yet, so if it doesn't make sense ignore my comment pls.

**Comment by szymonlesisz** [comment]
@martinboehm i would rather move this router check and add it to condition in discoveryMiddleware, maybe it's out of discovery context but at least it will be in one place:
https://github.com/trezor/trezor-suite/blob/develop/packages/suite/src/middlewares/wallet/discoveryMiddleware.ts#L26-L35

File: packages/suite/src/components/suite/modals/index.tsx [hunk 2]
```diff
@@ -68,11 +68,9 @@ const getDeviceContextModal = (props: Props) => {
         // T1 firmware
         case UI.INVALID_PIN:
             return <PinInvalid device={device} />;
-
```
Comment (szymonlesisz) [inline packages/suite/src/components/suite/modals/index.tsx]: unnecessary changes, pls revert this file

File: packages/suite/src/middlewares/wallet/discoveryMiddleware.ts [hunk 3]
```diff
@@ -34,6 +35,19 @@ const discoveryMiddleware = (api: MiddlewareAPI<Dispatch, AppState>) => (next: D
         return action;
     }
 
+    // temporary workaround - do not show backup modal in function call getAddress() in exchanges
+    // see useCoinmarketExchangeForm (line 104)
+    const isCoinmarketExchange = prevState.router.route?.name === 'wallet-coinmarket-exchange';
+
+    if (
```
Comment (szymonlesisz) [inline packages/suite/src/middlewares/wallet/discoveryMiddleware.ts]: add this condition to line 30:
if (action.type === UI.REQUEST_CONFIRMATION && (discoveryIsRunning || isCoinmarketExchange))

**Review by szymonlesisz — CHANGES_REQUESTED** [review]

**Review by szymonlesisz — APPROVED** [review]
